### PR TITLE
GPC-620: GroPullFile and GroPublishRecord auditing

### PR DESCRIPTION
**Pull request by hannah-leong**

### CODE REVIEW

File: lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/GroPullFile.java [hunk 1]
```diff
@@ -79,27 +87,38 @@ private void transferFile(String xmlBucket) {
             client.connect(host);
             client.authPublickey(username, privateKeyProvider);
 
-            logger.info("Pulling file {} from GRO", groFileName);
```
Comment (CarlyG55) [inline lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/GroPullFile.java:82]: I think we still want this logging in place - it fulfils a bit of a different purpose to the audit queue

File: lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/GroPullFile.java [hunk 2]
```diff
@@ -79,27 +87,38 @@ private void transferFile(String xmlBucket) {
...
             try (var sftpClient = client.newSFTPClient()) {
                 var resources = sftpClient.ls(sourceDir);
-                logger.info(resources);
```
Comment (CarlyG55) [inline lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/GroPullFile.java:85]: You can leave this one deleted though - this got put in just before all of the hmpo mess and then I forgot to take it out again 😅

File: lambdas/gro-publish-record/src/main/java/uk/gov/di/data/lep/GroPublishRecord.java [hunk 3]
```diff
@@ -97,4 +102,18 @@ private void postRecordToLifeEvents(GroJsonRecord event, String authorisationTok
             throw new GroApiCallException("Failed to send GRO record request", e);
         }
     }
+
+    @Tracing
+    private void audit(GroJsonRecordWithHeaders event) throws JsonProcessingException {
```
[the remark this answers is not in the document]
Comment (hannah-leong) [reply lambdas/gro-publish-record/src/main/java/uk/gov/di/data/lep/GroPublishRecord.java:107]: That's there for the `objectMapper.writeValueAsString` - if I put that line (and following) inside the try...catch then I can't get `auditData.eventName()` for the logs.

[the remark this answers is not in the document]
Comment (CarlyG55) [reply lambdas/gro-publish-record/src/main/java/uk/gov/di/data/lep/GroPublishRecord.java:107]: Hm I see - I still think we should be wrapping this `objectMapper` call in a try..catch block for consistency of errors. If it comes to it, given `auditData.eventName()` is always going to be the same in this `audit` function, I don't think it really matters that much if we just hardcode it

File: lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/GroPullFile.java [hunk 4]
```diff
@@ -80,26 +88,39 @@ private void transferFile(String xmlBucket) {
...
                 if (sourceFileSearch.isEmpty()) {
                     throw new GroSftpException(
-                            String.format("File: %s not found on GRO SFTP Server in directory: %s", groFileName, sourceDir)
+                        String.format("File: %s not found on GRO SFTP Server in directory: %s", groFileName, sourceDir)
                     );
                 }
 
-                try (var file = sftpClient.open(sourceFileSearch.get().getPath(), EnumSet.of(OpenMode.READ));
-                    var fileStream = file.new RemoteFileInputStream()) {
-                    logger.info("Uploading file {} to S3", groFileName);
```
Comment (CarlyG55) [inline lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/GroPullFile.java:96]: missed this logger line 🙂

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**